### PR TITLE
feat(optimize): add MPS HSL panic loss metrics

### PR DESCRIPTION
**Pull request by enarjord**
## Summary
- add the six resting-limit HSL panic-loss metrics to GPU scoring and limits for the supported one-sided single-coin EMA Anchor and Trailing Martingale topology
- tag only HSL panic fills, capture pre-fill episode equity, aggregate per-fill net losses, and finalize per-episode loss drawdowns at Rust's second-flat-confirmation boundary
- preserve coin-mode zero semantics for halt-to-restart loss and normalize pside/unified loss by starting balance
- keep exact Rust validation and drift gates authoritative

## TM parity fixes discovered by this slice
- HSL panic closes bypass recursive close-grid reconstruction and remain one exclusive full-position order
- HSL panic closes bypass the ordinary realized-loss gate, matching Rust

## Still fail closed
- HSL market panic execution
- finite rolling HSL PnL lookbacks
- dual-side and multi-coin HSL, including per-coin HSL overrides
- HSL strategy-equity EMA/recovery-distribution metrics

## Validation
- `python -m pytest -q tests/optimization`
- targeted GPU metric/backend tests (344 passed)
- physical M3: `python -m pytest -q tests/optimization/test_gpu_mps.py` (184 passed)
- physical long/short EMA/TM HSL panic cases with recursive TM closes and `max_realized_loss_pct=0`
- `cargo fmt --check`
- `cargo check --tests`
- `cargo test --no-default-features` (262 passed)
- rebuilt and runtime-stamped the Python Rust extension before integration/MPS tests

## Scope
Additive GPU optimizer support only. CPU optimization, live trading, and ordinary Rust backtesting paths are unchanged.

### CODE REVIEW

**Comment by enarjord** [comment]
@codex review\n\nPlease review exact head df6d9ea07, focusing on Rust parity for panic-fill attribution, pre-fill equity capture, episode finalization, coin-versus-pside halt loss, TM exclusive panic execution, scalar contracts, and fail-closed boundaries.

File: src/optimization/gpu/metrics.py [hunk 1]
```diff
@@ -613,6 +627,39 @@ def value(name: str):
     }
 
 
+def _hard_stop_panic_loss_metrics(out: dict, run) -> dict:
+    """Reduce resting-limit panic-fill losses using the Rust HSL formulas."""
+
+    reference = out["max_dd"].to(torch.float64)
+    zeros = torch.zeros_like(reference)
+    if "hsl_panic_close_loss_sum" not in out:
+        return {name: zeros for name in _HARD_STOP_PANIC_LOSS_METRICS}
```
[the remark this answers is not in the document]
Comment (enarjord) [reply src/optimization/gpu/metrics.py]: Fixed in 57acc20af. The single-coin proxy now retains all 25 directional HSL lifecycle and panic-loss raw outputs before reduction. Both reducers fail closed when directional fields are absent, and multi-coin HSL metric requests reject before dispatch. Added end-to-end proxy-retention, missing-output, and topology regression tests; full tests/optimization is green.

**Comment by enarjord** [comment]
@codex review\n\nP1 fixed. Please review exact head 57acc20af, including directional HSL output retention, fail-closed missing-output behavior, and multi-coin metric topology rejection.
